src/catalog/pgduckdb_table.cpp

```diff
@@ -47,14 +47,25 @@ PostgresTable::PopulateColumns(CreateTableInfo &info, Oid relid, Snapshot snapsh
 		auto duck_type = pgduckdb::ConvertPostgresToDuckColumnType(attr);
 		info.columns.AddColumn(duckdb::ColumnDefinition(col_name, duck_type));
 		/* Log column name and type */
-		elog(DEBUG3, "-- (DuckDB/PostgresHeapBind) Column name: %s, Type: %s --", col_name.c_str(),
+		elog(DEBUG3, "(DuckDB/SetTableInfo) Column name: %s, Type: %s --", col_name.c_str(),
 		     duck_type.ToString().c_str());
 	}
 
 	RelationClose(rel);
 	return true;
 }
 
+Cardinality
+PostgresTable::GetTableCardinality(Oid relid) {
+	auto rel = RelationIdGetRelation(relid);
```

Comment:
I think this probably needs the `DuckdbProcessLock`.

Reply:
Let me give another pass through code; I'm seeing this `RelationIdGetRelation` used on few other places (unprotected by lock) maybe it is worth checking if this function can be used in only one function and than we would pass pointer to its data.

Reply:
@Tishj can you look into latest commit it should centralize opening/closing relation for execution